spec/datadog/appsec/contrib/http/faraday_spec.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require 'datadog/appsec/spec_helper'
+require 'faraday'
+
+RSpec.describe 'AppSec Faraday integration' do
+  let(:waf_response_double) { double(match?: false) }
+  let(:active_context) { stub_const('Datadog::AppSec::Context', double(run_rasp: waf_response_double)) }
```

Comment:
I don't think it's a good idea, it allows us to craft what ever interface we want without any connection to the real code.

Instead, I would suggest this

```ruby
let(:context) { instance_double(Datadog::AppSec::Context, run_rasp: waf_response) }
let(:waf_response) { instance_double(Datadog::AppSec::SecurityEngine::Result, match?: false) }
```

_P.S I don't think that in tests it makes sense to add "something_double" to variables, because we are in tests and it should not matter until they behave the same_

Reply:
totally agree